eng/common/scripts/Delete-RemoteBranches.ps1

```diff
@@ -50,12 +71,37 @@ foreach ($branch in $branches)
     continue
   }
 
-  LogDebug "Branch [ $branchName ] in repo [ $RepoName ] has no associated open Pull Request. Deleting Branch"
-  try{
-    Remove-GitHubSourceReferences -RepoOwner $RepoOwner -RepoName $RepoName -Ref ($branch.Remove(0,5)) -AuthToken $AuthToken
-  }
-  catch {
-    LogError "Remove-GitHubSourceReferences failed with exception:`n$_"
-    exit 1
+  LogDebug "Branch [ $branchName ] in repo [ $RepoName ] has no associated open Pull Request."
+  # Allow to override the the var $LastCommitOlderThan
+  if ($LastCommitOlderThan -or $env:LASTCOMMITDATEOVERRIDE) {
+    $LastCommitOlderThan = If (!$LastCommitOlderThan) { $env:LASTCOMMITDATEOVERRIDE } Else { $LastCommitOlderThan } 
+    LogDebug "Deleteing the branch which last commit older than the date $LastCommitOlderThan."
+    
+    if (!$res.object -or !$res.object.url) {
+      LogDebug "No commit url returned from response. Skipping... "
+      continue
+    }
+    $GetCommitSHAUrl = $res.object.url
+    $responseJson = HandleGHAPICall -api $GetCommitSHAUrl
+    if ($null -eq $responseJson) {
+      LogError "Error calling the commit SHA API. Exiting..."
+      exit 1
+    }
+    
+    $commitResponse = ConvertFrom-Json $responseJson
+    if (!$commitResponse.author -or !$commitResponse.author.date) {
+      LogDebug "No date returned from the commit sha. Skipping..."
+    }
+    if ($commitResponse.author.date -gt $LastCommitOlderThan) {
```

Comment:
Why would the date be on the author object?

Reply:
OK I think I see why this is the case based on https://docs.github.com/en/rest/git/commits. Although I think I would use the committer.date instead of author.date